packages/next/src/server/app-render/create-component-tree.tsx

```diff
@@ -641,25 +641,26 @@ async function createComponentTreeInternal({
           )
 
           segmentNode = (
-            <NotFoundBoundary
+            <HTTPErrorFallbackBoundary
               key={cacheNodeKey}
-              notFound={
+              notFound={[
                 <>
                   {layerAssets}
                   {notfoundClientSegment}
-                </>
-              }
+                </>,
+                null,
+              ]}
             >
               {layerAssets}
               {clientSegment}
-            </NotFoundBoundary>
+            </HTTPErrorFallbackBoundary>
           )
         } else {
           segmentNode = (
-            <NotFoundBoundary key={cacheNodeKey}>
+            <HTTPErrorFallbackBoundary key={cacheNodeKey}>
```

Comment:
This boundary seems can be changed to fragment? as it's doing nth 🤔

Reply:
Yeah I left it in in an earlier refactor b/c I wasn't sure about the inner implementation and this was semantically equivalent. But since it just renders a passthrough when there is no NotFound we may as well eliminate it. That said, is it ever possible for NotFound to not exist? isn't there always a default one?

Reply:
This NotFound is the segment level component, so it's not always defined, i changed it to Fragment